src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1187 @@
+                        return $radioElement->getFirstSelectedOption()->getAttribute('value');
+                    } catch (NoSuchElementException $e) {
+                        return null;
+                    }
+
+                case $widgetType === 'checkbox':
+                    // WebDriverCheckboxes is not suitable since it _always_ behaves as a group
+                    return $element->isSelected() ? $element->getAttribute('value') : null;
+
+                case $widgetType === 'select':
+                    $selectElement = new WebDriverSelect($element);
+                    $selectedOptions = array_map(
+                        static fn(WebDriverElement $option) => $option->getAttribute('value'),
+                        $selectElement->getAllSelectedOptions()
+                    );
+                    return $selectElement->isMultiple() ? $selectedOptions : ($selectedOptions[0] ?? '');
+
+                default:
+                    return $this->getWebDriver()->isW3cCompliant()
+                        ? $element->getDomProperty('value')
+                        : $this->executeJsOnElement($element, 'return arguments[0].value');
+            }
+        } catch (Throwable $e) {
+            throw new DriverException("Cannot retrieve $widgetType value: {$e->getMessage()}", 0, $e);
+        }
+    }
+
+    public function setValue(
+        #[Language('XPath')]
+        string $xpath,
+        $value
+    ): void {
+        $element = $this->findElement($xpath);
+        $elementName = strtolower($element->getTagName() ?? '');
+
+        switch ($elementName) {
+            case 'textarea':
+                if (!is_string($value)) {
+                    throw new DriverException('Textarea value must be a string');
+                }
+                $element->clear();
+                $element->sendKeys($value);
+                break;
+
+            case 'select':
+                if (is_array($value)) {
+                    $this->deselectAllOptions($element);
+                    foreach ($value as $option) {
+                        $this->selectOptionOnElement($element, $option, true);
+                    }
+                    return;
+                }
+                $this->selectOptionOnElement($element, (string)$value);
+                return;
+
+            case 'input':
+                $elementType = strtolower((string)$element->getAttribute('type'));
+                switch ($elementType) {
+                    case 'submit':
+                    case 'image':
+                    case 'button':
+                    case 'reset':
+                        $message = 'Cannot set value on element with XPath "%s" as it is not a select, textarea or textbox';
+                        throw new DriverException(sprintf($message, $xpath));
+
+                    case 'color':
+                        if (!is_string($value)) {
+                            throw new DriverException('Color value must be a string');
```

Comment:
This case should be covered by tests in the HTML5Test of the driver testsuite (similar to the tests I added recently in SelectTest and RadioTest for similar bad usages).

Reply:
Done: https://github.com/minkphp/driver-testsuite/pull/78